### PR TITLE
Implement create timetable, and delete timetable by ID endpoint

### PR DESCRIPTION
**Pull request by soumitradev**

### CODE REVIEW

File: src/controllers/timetable/deleteTimetable.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    if (!author) {
+      return res.json({ message: "unregistered user" });
+    }
+
+    const id: number = parseInt(req.params.id);
+
+    let timetable: Timetable | null = null;
+
+    try {
```
Comment (soumitradev) [inline src/controllers/timetable/deleteTimetable.ts:67]: This again feels like an unnecessary check, does delete not return `null` or error out with a specific error code if such a timetable doesn't exist?

File: src/controllers/timetable/deleteTimetable.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    if (!timetable) {
+      return res.status(404).json({ message: "timetable not found" });
+    }
+
+    let owns: boolean = false;
+
+    try {
+      owns =
```
Comment (soumitradev) [inline src/controllers/timetable/deleteTimetable.ts:85]: This again feels like an unnecessary check, does delete not return `null` or error out with a specific error code if such a timetable doesn't exist (i.e. where author has given email)?

File: src/controllers/timetable/deleteTimetable.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+      res.status(500).json({ message: "Internal Server Error" });
+    }
+
+    if (!owns) {
+      return res.status(403).json({ message: "user does not own timetable" });
+    }
+
+    try {
+      await timetableRepository.delete({ id });
```
Comment (soumitradev) [inline src/controllers/timetable/deleteTimetable.ts:103]: Migrate this to builder pattern

File: src/utils/zodValidateBody.ts
Comment (soumitradev) [inline src/utils/zodValidateBody.ts:1]: File name seems misleading: the function validates more than just the request body.

**Review by soumitradev — COMMENTED** [review]
Since I cannot mark this review as "Request changes" (since I am the author of this PR now), consider my comments as requests for changes.

File: src/controllers/timetable/createTimetable.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        }
+      ),
+  }),
+});
+
+export const createTimeTableValidator = validate(dataSchema);
+
+export const createTimetable = async (req: Request, res: Response) => {
+  try {
+    let author: User | null = null;
```
[the remark this answers is not in the document]
Comment (majimearun) [reply src/controllers/timetable/createTimetable.ts:35]: its required as we need to handle null before using its properties (`author.degrees` in this case), so i thought might as well explicitly set a status and message.

File: src/controllers/timetable/deleteTimetable.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        message: "invalid id",
+      }),
+  }),
+});
+
+export const deleteTimeTableValidator = validate(dataSchema);
+
+export const deleteTimetable = async (req: Request, res: Response) => {
+  try {
+    let author: User | null = null;
```
[the remark this answers is not in the document]
Comment (majimearun) [reply src/controllers/timetable/deleteTimetable.ts:46]: its required as we need to handle null before using its properties (`author.id` in this case), so i thought might as well explicitly set a status and message.

File: src/utils/zodValidateBody.ts
Comment (majimearun) [reply src/utils/zodValidateBody.ts:1]: changed name to `zodValidateRequest.ts`

File: src/controllers/timetable/createTimetable.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    const sections: Section[] = [];
+    const timings: string[] = [];
+    const midsemTimes: Date[] = [];
+    const compreTimes: Date[] = [];
+    const warnings: string[] = [];
+    const createdAt: Date = new Date();
+    const lastUpdated: Date = new Date();
+
+    try {
+      const timetable: Timetable = timetableRepository.create({
```
[the remark this answers is not in the document]
Comment (majimearun) [reply src/controllers/timetable/createTimetable.ts]: shifted

File: src/controllers/timetable/deleteTimetable.ts (hunk 2 above)
Comment (majimearun) [reply src/controllers/timetable/deleteTimetable.ts:85]: repeated comment so just marking as resolved

File: src/controllers/timetable/deleteTimetable.ts (hunk 1 above)
Comment (majimearun) [reply src/controllers/timetable/deleteTimetable.ts:67]: again, the check is before handling null before using its id (which is required by the builder pattern)

File: src/controllers/timetable/deleteTimetable.ts (hunk 3 above)
Comment (majimearun) [reply src/controllers/timetable/deleteTimetable.ts]: shifted

**Pull request merged**